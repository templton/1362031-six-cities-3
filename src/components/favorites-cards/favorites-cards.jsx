import React, {PureComponent, Fragment} from "react";
import {connect} from "react-redux";
import PropTypes from "prop-types";
import {selectFavourites} from "../../store/favourites/selectors";
import {Operation as FavouritesOperation} from "../../store/favourites/reducers";
import PlacesList from "../places-list/places-list";


class FavoritesCards extends PureComponent {

  componentDidMount() {
    const {loadFavourites} = this.props;
    loadFavourites();
  }

  render() {
    const {hotels} = this.props;

    const sities = Array.from(new Set(hotels.map((item)=>{
      return item.city.name;
    })));

    if (sities.length === 0) {
      return (
        <Fragment>
          <main className="page__main page__main--favorites page__main--favorites-empty">
            <div className="page__favorites-container container">
              <section className="favorites favorites--empty">
                <h1 className="visually-hidden">Favorites (empty)</h1>
                <div className="favorites__status-wrapper">
                  <b className="favorites__status">Nothing yet saved.</b>
                  <p className="favorites__status-description">Save properties to narrow down search or plan yor future
                    trips.</p>
                </div>
              </section>
            </div>
          </main>
          <footer className="footer">
            <a className="footer__logo-link" href="main.html">
              <img className="footer__logo" src="img/logo.svg" alt="6 cities logo" width="64" height="33"/>
            </a>
          </footer>
        </Fragment>
      );
    }

    return (
      <main className="page__main page__main--favorites">
        <div className="page__favorites-container container">
          <section className="favorites">
            <h1 className="favorites__title">Saved listing</h1>
            <ul className="favorites__list">
              {sities.map((city, index)=>{
                return (
                  <li key={index} className="favorites__locations-items">
                    <div className="favorites__locations locations locations--current">
                      <div className="locations__item">
                        <a className="locations__item-link" href="#">
                          <span>{city}</span>
                        </a>
                      </div>
                    </div>
                    <div className="favorites__places">
                      <PlacesList
                        imgHeight={110}
                        imgWidth={150}
                        places={hotels.filter((item)=>(item.city.name === city))}
                        cardClass={`favorites__card`}
                        imageWrapperClass="favorites__image-wrapper"/>
                    </div>
                  </li>
                );
              })}
            </ul>
          </section>
        </div>
      </main>
    );
  }
}

FavoritesCards.propTypes = {
  hotels: PropTypes.array,
  loadFavourites: PropTypes.func.isRequired
};

const mapStateToProps = (state) => ({
  hotels: selectFavourites(state)
});

const mapDispatchToProps = ({
  loadFavourites: FavouritesOperation.loadFavourites
});

export {FavoritesCards};
export default connect(mapStateToProps, mapDispatchToProps)(FavoritesCards);
