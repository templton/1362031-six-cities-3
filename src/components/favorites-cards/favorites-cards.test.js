import React from "react";
import renderer from "react-test-renderer";
import {FavoritesCards} from "./favorites-cards";

describe(`FavoritesCards render`, () => {

  it(`FavoritesCards render empty page`, () => {
    const tree = renderer
      .create(<FavoritesCards hotels={[]} loadFavourites={()=>([])}/>).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
