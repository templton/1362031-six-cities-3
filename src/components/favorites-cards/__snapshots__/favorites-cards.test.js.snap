// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoritesCards render FavoritesCards render empty page 1`] = `
Array [
  <main
    className="page__main page__main--favorites page__main--favorites-empty"
  >
    <div
      className="page__favorites-container container"
    >
      <section
        className="favorites favorites--empty"
      >
        <h1
          className="visually-hidden"
        >
          Favorites (empty)
        </h1>
        <div
          className="favorites__status-wrapper"
        >
          <b
            className="favorites__status"
          >
            Nothing yet saved.
          </b>
          <p
            className="favorites__status-description"
          >
            Save properties to narrow down search or plan yor future trips.
          </p>
        </div>
      </section>
    </div>
  </main>,
  <footer
    className="footer"
  >
    <a
      className="footer__logo-link"
      href="main.html"
    >
      <img
        alt="6 cities logo"
        className="footer__logo"
        height="33"
        src="img/logo.svg"
        width="64"
      />
    </a>
  </footer>,
]
`;
